pandas/_libs/src/ujson/python/JSONtoObj.c

```diff
@@ -108,6 +108,9 @@ JSOBJ Object_npyNewArray(void *prv, void *_decoder) {
     if (decoder->curdim <= 0) {
         // start of array - initialise the context buffer
         npyarr = decoder->npyarr = PyObject_Malloc(sizeof(NpyArrContext));
+        if (npyarr == NULL) {
```

Comment:
I think this is a bit redundant with L116, though I find the `npyarr == NULL` approach more explicit. I also think the position of this is better than L116 but need to combine these pieces of code